### ISSUE TITLE
`SS_SHOULD_BE_STATIC` missed when a `final` constant field is initialized in an instance-initializer block instead of at the declaration

### ISSUE THREAD

**Issue by Chordrain** [spotbugs/spotbugs#4136]
### SpotBugs version

SpotBugs 4.9.8 (JDK 21, javac 21, source/target 21)

### Bug pattern

[SS_SHOULD_BE_STATIC](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#ss-unread-field-should-this-field-be-static-ss-should-be-static)

### Description

An unread `final` field whose value is a compile-time constant String is flagged when written at the declaration site (`final String staticIsMissingPath = "/search";`) but not when the identical constant is assigned in an instance-initializer block (`final String staticIsMissingPath; { staticIsMissingPath = "/search"; }`).

The constructor bytecode is **identical** for both forms (same `ldc "/search"; putfield staticIsMissingPath`), and the field is unread in both. The only class-file difference is the `ConstantValue` attribute, present only for the declaration-site form:

```
BEFORE  field staticIsMissingPath:  ConstantValue: String /search   <-- present
AFTER   field staticIsMissingPath:  (no ConstantValue)               <-- absent
<init>  (both):  ... ldc "/search"; putfield staticIsMissingPath ...  <-- identical
```

Note that only the `final` field triggers `SS_SHOULD_BE_STATIC`; non-`final` fields in the same class (e.g. `String fileName = "...";`) are not reported in either form, confirming the report keys off the constant/`ConstantValue` status of the `final` field. Because the relocated form drops `ConstantValue` while remaining unread and constant-valued, the detector misses it — a false negative.

### Reproducer

```java
// BEFORE
public class HardcodedURICheckSample {
    final String staticIsMissingPath = "/search";   // reported: should be static
}

// AFTER
public class HardcodedURICheckSample {
    final String staticIsMissingPath;
    { staticIsMissingPath = "/search"; }             // same constant, unread -> NOT reported
}
```

### Expected behavior

An unread `final` field assigned a compile-time constant should be flagged whether the constant is written at the declaration or in an instance-initializer block; the two forms produce identical constructor bytecode and leave the field equally unread and static-eligible.

### Actual behavior

- BEFORE: **1** report — `Unread field: HardcodedURICheckSample.staticIsMissingPath; should this field be static?`
- AFTER:  **0** reports.

**Comment by omer-cengel** [comment]
Hi!

I opened a PR https://github.com/spotbugs/spotbugs/pull/4207 that fixes this issue. When you have time, could you please take a look and review it? Thanks!
